src/daemon/daemonlocalserverconnection.cpp

```diff
@@ -160,3 +171,59 @@ void DaemonLocalServerConnection::write(const QJsonObject& obj) {
   m_socket->write(QJsonDocument(obj).toJson(QJsonDocument::Compact));
   m_socket->write("\n");
 }
+
+bool DaemonLocalServerConnection::isCallerAuthorized(const QString& command) {
+#if defined(MZ_MACOS)
+  uid_t uid;
+  gid_t gid;
+  int result = getpeereid(m_socket->socketDescriptor(), &uid, &gid);
+  if (result == -1) {
+    logger.error() << "Unable to determine if caller is authorized."
+                      "Assuming unauthorized. Error:"
+                   << errno;
+    return false;
+  }
+
+  if (uid == 0) {
+    logger.debug() << "Command issued by root.";
+    // Case: This is root calling, root is always authorized.
+    return true;
+  }
+
+  Daemon::State state = Daemon::instance()->state();
+  if (state != Daemon::Inactive) {
```

Comment:
Why do we check for `if (state != Daemon::Inactive)` here? Regardless of the Daemon state, shouldn't we check whether or not a caller is authorized?

Reply:
Not in the case of `status` or `activate`.